### PR TITLE
[Brent][FMS] Add config for changing text on submit ggw payment

### PR DESCRIPTION
**Pull request by MorayMySoc**
Change text on button for Brent payment by adding config override for button text

https://3.basecamp.com/4020879/buckets/28996137/todos/5838639014#__recording_5996609233

[skip changelog]

### CODE REVIEW

File: perllib/FixMyStreet/App/Form/Waste/Garden.pm [hunk 1]
```diff
@@ -110,7 +110,11 @@ has_page summary => (
         $data->{cost_pa} = $cost_pa / 100;
         $data->{display_total} = $total / 100;
 
-        return {};
+        my $button_text = 'Continue to payment';
+        if ($form->{c}->cobrand->feature('waste_features') && $form->{c}->cobrand->feature('waste_features')->{text_for_waste_payment}) {
```
Comment (dracos) [inline perllib/FixMyStreet/App/Form/Waste/Garden.pm:114]: I think this doesn't need the first half? feature() will return undef and you can look up a key on that without warning (unless I've missed something).

**Review by dracos — CHANGES_REQUESTED** [review]
I don't think this does what's requested, which is to change the text only for staff who will be getting the PAYE form on the next page? This changes it for everyone, including normal users?

File: perllib/FixMyStreet/App/Form/Waste/Garden.pm (hunk 1 above)
Comment (MorayMySoc) [reply perllib/FixMyStreet/App/Form/Waste/Garden.pm:114]: Running test suite getting errors like '[error] Caught exception in FixMyStreet::App::Controller::Waste->form "Can't use an undefined value as a HASH reference at /home/moray/FixMyStreet/fixmystreet/perllib/FixMyStreet/App/Form/Waste/Garden/Renew.pm line 80."' so thought that was the issue

Comment (dracos) [reply perllib/FixMyStreet/App/Form/Waste/Garden.pm:114]: Fair enough, then!

Comment (dracos) [reply perllib/FixMyStreet/App/Form/Waste/Garden.pm:114]: Though I don't see why it wouldn't: ` perl -we'use strict; my $d; my $e = $d->{123};` doesn't error.

Comment (dracos) [reply perllib/FixMyStreet/App/Form/Waste/Garden.pm:114]: Ah, so it's okay if you put it in a variable first, but not if you try and use it directly.

File: perllib/FixMyStreet/App/Form/Waste/Garden.pm [hunk 2]
```diff
@@ -251,6 +252,7 @@ has_field continue_review => (
 
 has_field submit => (
     type => 'Submit',
+    value => 'Continue to payment',
```
Comment (dracos) [inline perllib/FixMyStreet/App/Form/Waste/Garden.pm]: Is this needed, your code change doesn't seem to affect this always being set above?

File: perllib/FixMyStreet/App/Form/Waste/Garden.pm [hunk 3]
```diff
@@ -128,8 +128,9 @@ has_page summary => (
         $data->{display_total} = $total / 100;
 
         my $button_text = 'Continue to payment';
-        if ($form->{c}->cobrand->feature('waste_features') && $form->{c}->cobrand->feature('waste_features')->{text_for_waste_payment}) {
-            $button_text = $form->{c}->cobrand->feature('waste_features')->{text_for_waste_payment};
+        my $features = $form->{c}->cobrand->feature('waste_features');
```
Comment (dracos) [inline perllib/FixMyStreet/App/Form/Waste/Garden.pm:131]: $c to be consistent with line below? :)

File: t/app/controller/waste_brent_garden.t [hunk 4]
```diff
@@ -281,6 +282,7 @@ FixMyStreet::override_config {
             # external redirects make Test::WWW::Mechanize unhappy so clone
             # the mech for the redirect
             my $mech2 = $mech->clone;
+            $mech2->content_contains('Continue to payment', 'Waste features text_for_waste_payment used');
```
Comment (dracos) [inline t/app/controller/waste_brent_garden.t:285]: Description isn't right here no (but is testing right thing)

File: perllib/FixMyStreet/App/Form/Waste/Garden.pm (hunk 2 above)
Comment (dracos) [reply perllib/FixMyStreet/App/Form/Waste/Garden.pm]: I don't understand, it was removed then added according to the git log but isn't present in the files view. But I think can be removed!

**Review by dracos — CHANGES_REQUESTED** [review]
Couple of tiny changes only :) I wondered whether instead of a new feature it used whether $c->stash->{staff_payments_allowed} was 'paye', but this way is a bit more flexible, I guess.

**Review by dracos — APPROVED** [review]

**Pull request merged**